### PR TITLE
fix: tokens loader use text and add esm output

### PR DESCRIPTION
**Pull request by jeasonstudio**
* fix esbuild text loader, its `text` but not `file`.
* add `esm` output type.

### CODE REVIEW

**Comment by jeasonstudio** [comment]
Currently, there is an error in the loader configuration of esbuild, which causes the `.tokens` file to not be loaded correctly during the building process for browser targets. This results in incorrect tokenization. Please review this PR when you have time. Thanks.
@Janther @mattiaerre @naddison36 @fvictorio

File: package.json [hunk 1]
```diff
@@ -14,9 +14,10 @@
...
   "scripts": {
     "antlr": "antlr4ts -visitor antlr/Solidity.g4 -o src",
-    "build:browser": "esbuild src/index.ts --outfile=dist/index.iife.js --bundle --loader:.tokens=file --sourcemap --format=iife --global-name=SolidityParser --define:__dirname=true --define:BROWSER=true --inject:./process-shim.js",
+    "build:browser": "esbuild src/index.ts --outfile=dist/index.iife.js --bundle --loader:.tokens=text --sourcemap --format=iife --global-name=SolidityParser --define:__dirname=true --define:BROWSER=true --inject:./process-shim.js",
+    "build:esm": "esbuild src/index.ts --outfile=dist/index.esm.js --bundle --loader:.tokens=text --sourcemap --format=esm --global-name=SolidityParser --define:__dirname=true --define:BROWSER=true --inject:./process-shim.js",
     "build:node": "esbuild src/index.ts --outfile=dist/index.cjs.js --bundle  --loader:.tokens=file --sourcemap --format=cjs --platform=node --target=node12",
-    "build": "npm run build:node && npm run build:browser && npm run generate-types && npm run copy-files",
+    "build": "npm run build:node && npm run build:browser && npm run build:browser && npm run generate-types && npm run copy-files",
```
Comment (holic) [inline package.json:20]:
```suggestion
    "build": "npm run build:node && npm run build:browser && npm run build:esm && npm run generate-types && npm run copy-files",
```

Comment (jeasonstudio) [reply package.json]: Fixed

**Comment by fvictorio** [comment]
Closing because of [this](https://github.com/solidity-parser/parser/pull/85#issuecomment-1845287242). Happy to continue the discussion in [the issue I opened](https://github.com/solidity-parser/parser/issues/102).

**Pull request closed without merging**